src/core/HyperlanePermissionlessDeployer.ts

```diff
@@ -64,11 +64,20 @@ export function getArgs(multiProvider: MultiProvider) {
     .coerce('key', assertBytes32)
     .demandOption('key')
     .middleware(
-      assertBalances(multiProvider, (argv) => argv.remotes.concat(argv.local)),
+      assertBalances(multiProvider, (argv) =>
+        argv.remotes
+          .concat(argv.local)
+          .filter((chain: string) => !argv.skipDeployTo.includes(chain)),
+      ),
     )
     .describe('write-agent-config', 'Whether or not to write agent config')
     .default('write-agent-config', true)
-    .boolean('write-agent-config').argv;
+    .boolean('write-agent-config')
+    .describe('skip-deploy-to', 'Chains to skip deploying directly to')
```

Comment:
Would it be easier to just auto-skip non-evm chains?

Reply:
The foreignDeployment stuff in the RouterConfig would still be needed to know what router to enroll, but now that we'll have the protocol in chain metas I'm supportive of just skipping non-evm chains instead of `skip-deploy-to`